multinet/api/models/workspace.py

```diff
@@ -20,48 +22,134 @@ def create_default_arango_db_name():
-    def owners(self):
-        return get_users_with_perms(self, only_with_perms_in=['owner'])
-
-    def set_owners(self, new_owners):
-        old_owners = get_users_with_perms(self, only_with_perms_in=['owner'])
-
-        removed_owners = []
-        added_owners = []
-
-        # Remove old owners
-        for old_owner in old_owners:
-            if old_owner not in new_owners:
-                remove_perm('owner', old_owner, self)
-                removed_owners.append(old_owner)
-
-        # Add new owners
-        for new_owner in new_owners:
-            if new_owner not in old_owners:
-                assign_perm('owner', new_owner, self)
-                added_owners.append(new_owner)
-
-        # Return the owners added/removed so they can be emailed
-        return removed_owners, added_owners
-
-    def add_owner(self, new_owner):
-        old_owners = get_users_with_perms(self, only_with_perms_in=['owner'])
-        if new_owner not in old_owners:
-            assign_perm('owner', new_owner, self)
-
-    def remove_owner(self, owner):
-        owners = get_users_with_perms(self, only_with_perms_in=['owner'])
-        if owner in owners:
-            remove_perm('owner', owner, self)
+    def owner(self):
+        """
+        Return the workspace owner.
+
+        Django-guardian allows for potentially more than one user with
+        the 'owner' permission, so we must take care to limit the number of owners ourselves by
+        using this property and Workspace.set_owner to handle assigning owners.
+        """
+        owners = list(
+            get_users_with_perms(self, only_with_perms_in=[WorkspacePermission.owner.name])
+        )
+        if len(owners) > 0:
+            return owners[0]
+        return None
+
+    @property
+    def maintainers(self):
+        return get_users_with_perms(self, only_with_perms_in=[WorkspacePermission.maintainer.name])
+
+    @property
+    def writers(self):
+        return get_users_with_perms(self, only_with_perms_in=[WorkspacePermission.writer.name])
+
+    @property
+    def readers(self):
+        return get_users_with_perms(self, only_with_perms_in=[WorkspacePermission.reader.name])
+
+    def get_user_permission(self, user: User) -> WorkspacePermission:
+        """
+        Get the object-level permission for a given user on this workspace.
+
+        In the event that there are more than one (not ideal), return the highest
+        ranking permission. Return None if the user has no permission for this workspace.
+        """
+        permission_names = get_user_perms(user, self)
+        hierarchichal_permissions = [
+            WorkspacePermission[name]
+            for name in permission_names
+            if name in WorkspacePermission.get_permission_codenames()
+        ]
+
+        if len(hierarchichal_permissions) == 0:
+            return None
+
+        max_permission = max(hierarchichal_permissions, key=lambda perm: perm.value)
+        return max_permission
```

Comment:
Python 3.4 added the [`default`](https://docs.python.org/3/library/functions.html#max) argument for empty iterables
```suggestion
        return max(hierarchichal_permissions, key=lambda perm: perm.value, default=None)
```